File: 2.js
// Day 2
// http://adventofcode.com/day/2

function part1(){
	var lines = document.getElementsByTagName("pre")[0].innerHTML.split("\n");
	var total = 0;
	for(var i = 0; i < lines.length; i++){
		if(lines[i].trim() == "")
			continue;
		var x = lines[i].split("x");
		var s = [x[0]*x[1], x[1]*x[2], x[2]*x[0]];
		var min = -1;
		s.forEach(function(a){
			if(a < min || min == -1)
				min = a;
		});
		total += (2 * parseInt(s[0]+s[1]+s[2]) + parseInt(min));
	}
	return total;
}

function part2(){
	var lines = document.getElementsByTagName("pre")[0].innerHTML.split("\n");
	var total = 0;
	for(var i = 0; i < lines.length; i++){
		if(lines[i].trim() == "")
			continue;
		var x = lines[i].split("x");
		x.sort(function(a, b){
			return a - b;
		});
		total += (2 * (parseInt(x[0]) + parseInt(x[1]))) + (x[0] * x[1] * x[2])
	}
	return total;
}

File: 2.py
#!/usr/bin/env python3
'''
    Day 2
    http://adventofcode.com/2015/day/2
'''

from itertools import combinations
from utils import get_input_filename
from functools import reduce

DAY_NUM = 2

# This does not feel very good
def part1():
    wrapping_paper = 0
    with open(get_input_filename(DAY_NUM)) as f:
        for line in f:
            sides = [reduce(lambda x, y: int(x) * int(y), combination) for combination in combinations(line.split('x'), 2)]
            wrapping_paper += min(sides)
            for side in sides:
                wrapping_paper += side * 2
    return wrapping_paper



if __name__ == '__main__':
    print(part1())
    # print(part2())
